### ISSUE TITLE
inequality syntax for datetime

### ISSUE THREAD

**Issue by marsupialtail** [pola-rs/polars#4970]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of polars.

### Issue Description

Possibly related to this: https://github.com/pola-rs/polars/issues/3965

If you run this query on TPCH SF1: https://github.com/pola-rs/tpch/blob/main/polars_queries/q3.py you will get wrong result. This is because polars treats the datetime inequalities larger than and smaller than as LTE and GTE. Get rid of the limit and count the number of rows in the result: a SQL engine will give 11620 rows while Polars gives 11758 rows.

### Reproducible Example

```python
as linked above.
```

### Expected Behavior

datetime inequalities should work as expected

### Installed Versions

<details>
``` 0.13.54 ```
</details>

**Comment by ritchie46** [comment]
Are you certain you operate on datetimes and not on strings?

**Comment by zundertj** [comment]
To expand on Richie's request, please follow the issue template guideline on how to construct an example:

```
Please follow [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) on how to provide a minimal, copy-pastable example. Include the (wrong) output if applicable.
```

In particular, your example is not minimal: there are 45 lines of code, there is an import from another module (which imports more modules) and from your question it reads the example datasets have over 11000 rows. Please make a small example, using just polars, with a small dataframe (if you can show it on a 1x1 dataframe, great!) crafted using code as the input, so that the bug is obvious.

In summary: we can help you, if you help us.

**Comment by marsupialtail** [comment]
Good point @ritchie46. When you do polars.read_csv from the lineitems table I didn't notice that the default datatype is string for the date columns. I didn't realize parse_dates is a separate option.
